src/distributions/other.rs

```diff
@@ -249,7 +250,7 @@ mod tests {
             '\u{ed692}',
             '\u{35888}',
         ]);
-        test_samples(&Alphanumeric, 'a', &['h', 'm', 'e', '3', 'M']);
+        test_samples(&Alphanumeric, 0, &[104, 109, 101, 51, 77]);
```

Comment:
Now we need `.map(..)` for distributions?

Possible I think, and in a way it makes sense. What do you think?

Reply:
> Now we need .map(..) for distributions?

How do you mean that? You need that if you want `char`, but I think that makes sense, because the conversion from `u8` to `char` is trivial, but the other direction is not. I think it makes more sense to see `Alphanumeric` as a distribution of bytes, because this type is more narrow, and it's unfortunate to throw away that compile-time knowledge by forcing a conversion to `char`.

If you prefer, we can also use this for the test:

```suggestion
        test_samples(&Alphanumeric, b'a', &[b'h', b'm', b'e', b'3', b'M']);
```

Reply:
No, I mean we could add a `map` method to `Distribution` just like `Iterator::map`. Not really needed for this PR, just an idea.

Reply:
At this point, we might as well implement `Iterator` for distributions, no?

Reply:
We do: the `.sample_iter(rng)` method. The RNG has to be attached somehow.

Reply:
I think it is preferable to convert the distribution into an iterator for such cases. This also supports all the other `Iterator` methods, without adding more API.